runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClusterState.java

```diff
@@ -53,20 +64,41 @@ public int size() {
     /**
      * See if cluster is ready. If cluster contains at least one node with state NOT_READY the cluster is not ready and
      * cluster state can't be used to find failures.
+     *
      * @return cluster status
      */
-    public boolean isReady(){
-        if (nodes.isEmpty()){
+    public boolean isReady() {
+        if (nodes.isEmpty()) {
             return false;
         }
 
         //if at least one node is not ready then entire cluster is not ready to provide correct information
         for (NodeState nodeState : nodes.values()) {
-            if (nodeState.getConnectivity().getType() == NodeConnectivityType.NOT_READY){
+            if (nodeState.getConnectivity().getType() == NodeConnectivityType.NOT_READY) {
                 return false;
             }
         }
 
         return true;
     }
+
+    public NodeConnectivity getLocalNodeConnectivity() {
+        Optional<NodeState> maybeNodeState = getNode(localEndpoint);
```

Comment:
Rename this to localNodeState

Reply:
the Optional value can be empty, that's why it's `maybe`, in the other case I would have this situation:
```
Optional<NodeState> nodeState = getNode(localEndpoint);
if(nodeState.isPresent()){
  NodeState ??? = nodeState.get();
} 
```